gramlib/grammar.ml

```diff
@@ -1548,24 +1548,29 @@ module Parsable = struct
       Stream.Failure ->
       let loc = get_loc () in
       restore ();
-      Ploc.raise loc (Stream.Error ("illegal begin of " ^ entry.ename))
+      Loc.raise ~loc (Stream.Error ("illegal begin of " ^ entry.ename))
     | Stream.Error _ as exc ->
-      let loc = get_loc () in restore (); Ploc.raise loc exc
+      let loc = get_loc () in restore ();
+      Loc.raise ~loc exc
     | exc ->
+      let exc,info = Exninfo.capture exc in
       let loc = Stream.count cs, Stream.count cs + 1 in
-      restore (); Ploc.raise (Ploc.make_unlined loc) exc
+      restore ();
+      let info = Loc.add_loc info (Ploc.make_unlined loc) in
```

Comment:
There is a small change in semantics here, we overwrite the old location if existing; indeed I think I should fix that.